overlord/state/state.go

```diff
@@ -341,6 +355,13 @@ func (s *State) Prune(pruneWait, abortWait time.Duration) {
 			delete(s.changes, chg.ID())
 		}
 	}
+	for tid, t := range s.tasks {
+		// TODO: this could be done more aggressively
+		if t.Change() == nil && t.SpawnTime().Before(pruneLimit) {
+			s.writing()
+			delete(s.tasks, tid)
```

Comment:
@chipaca it's a natural occurrence of any path that creates some tasks then discovers it needs to error out, it's ok to try to postpone making changes, but posting making tasks themselve acrosss multilple handlers is a pain, we could log it at debug level